test/Test_Statics.re

```diff
@@ -889,214 +792,172 @@ let tests = (
+                        FTemp.(
+                          Expectation({
+                            ana:
+                              ty_parens(
+                                ty_prod([
+                                  ty_int(),
+                                  ty_tup_label(ty_label("a"), ty_string()),
+                                ]),
+                              ),
+                            syn:
+                              ty_prod([
+                                ty_tup_label(ty_label("c"), ty_int()),
+                                ty_tup_label(ty_label("a"), ty_string()),
+                              ]),
+                          })
                         ),
-                        no_error_exp(Int(1)),
                       ),
                     ),
-                    no_error_exp(
-                      TupLabel(
-                        no_error_exp(Label("a")),
-                        no_error_exp(String("hello")),
+                  ),
+                ),
+              tuple(
+                ~ann=
+                  Some(
+                    Exp(
+                      Common(
+                        TupleLabelError({
+                          malformed_labels: [],
+                          duplicate_labels: [],
+                          invalid_labels: ["c"],
+                          typ:
+                            FTemp.(
+                              ty_prod([
+                                ty_tup_label(ty_label("c"), ty_int()),
+                                ty_tup_label(ty_label("a"), ty_string()),
+                              ])
+                            ),
+                        }),
                       ),
                     ),
-                  ]),
-                ),
+                  ),
+                [
+                  {
+                    tup_label(
+                      ~ann=
+                        Some(
+                          Exp(
+                            Common(
+                              TupleLabelError({
+                                malformed_labels: [],
+                                duplicate_labels: [],
+                                invalid_labels: ["c"],
+                                typ:
+                                  FTemp.(
+                                    ty_tup_label(ty_label("c"), ty_int())
+                                  ),
+                              }),
+                            ),
+                          ),
+                        ),
+                      error_exp(
+                        Exp(Common(NoType(InvalidLabel("c")))),
+                        Label("c"),
+                      ),
+                      int(1),
+                    );
+                  },
+                  tup_label(label("a"), string("hello")),
+                ],
               ),
             ),
-            no_error_exp(Bool(true)),
-          ),
+            bool(true),
+          )
```

Comment:
The test is significantly smaller and only 30 out of the remaining ~90 lines are the AST. The rest are the annotations.